include/deal.II/base/symmetric_tensor.h

```diff
@@ -1217,14 +1217,14 @@ namespace internal
       case 2:
         return (data[0] * sdata[0] +
                 data[1] * sdata[1] +
-                Number(2.) * data[2] * sdata[2]);
```

Comment:
I just checked a simple program and it is indeed as @bangerth says: You cannot multiply a complex float with a double constant `2.0`, at least not with gcc-6 and clang-3.7. But this means that many functions in `SymmetricTensor` have not been tested with complex floats, which is not really surprising because one does not often use `float` in the first place. I guess the best thing to do is to go with some temporary definition in the symmetric tensor. (Or to add the said constructors to `VectorizedArray` from the underlying scalar and cross fingers. It would make life a lot easier in a number of places.)